vertx/feign-vertx/src/main/java/feign/vertx/VertxHttpClient.java

```diff
@@ -43,28 +44,25 @@
  */
 @SuppressWarnings("unused")
 public final class VertxHttpClient {
-  private final HttpClient httpClient;
+  private final WebClient webClient;
   private final long timeout;
-  private final UnaryOperator<HttpClientRequest> requestPreProcessor;
+  private final UnaryOperator<HttpRequest<Buffer>> requestPreProcessor;
 
   /**
    * Constructor from {@link Vertx} instance, HTTP client options and request timeout.
    *
-   * @param vertx vertx instance
-   * @param options HTTP options
+   * @param webClient vertx WebClient
    * @param timeout request timeout
    * @param requestPreProcessor request pre-processor
    */
   public VertxHttpClient(
-      final Vertx vertx,
-      final HttpClientOptions options,
+      final WebClient webClient,
       final long timeout,
-      final UnaryOperator<HttpClientRequest> requestPreProcessor) {
```

Comment:
And lastly here too, keep old constructor, deprecate it, add new one

Reply:
Idem. Current version from core project was never released.